components/tools/OmeroWeb/omeroweb/webgateway/views.py

```diff
@@ -1463,6 +1474,60 @@ def get_thumb_url(iid):
 
 @login_required()
 @jsonp
+def get_thumbnails_json(request, w=None, conn=None, **kwargs):
+    """
+    Returns base64 encoded jpeg with the rendered thumbnail for images
+    'id'
+
+    @param request:     http request
+    @param w:           Thumbnail max width. 96 by default
+    @return:            http response containing base64 encoded thumbnails
+    """
+    if w is None:
+        w = 96
+    image_ids = get_longs(request, 'id')
+    logger.debug("Image ids: %r" % image_ids)
+    if len(image_ids) > settings.THUMBNAILS_BATCH:
+        return HttpJavascriptResponseServerError(
+            'Max %s thumbnails at a time.' % settings.THUMBNAILS_BATCH)
```

Comment:
The docs say "Make sure the size is not too big, otherwise you may excede limit request line" which would suggest that it is the main reason for the limit. If other reasons are more important then maybe the docs should be updated?
Why do we have to throw an error here? If a client asks for a page size larger than the PAGE setting, we don't throw an error.